.eslintrc

```diff
@@ -6,19 +6,57 @@
   },
   "extends": [
     "semistandard",
-    "eslint:recommended",
-    "plugin:prettier/recommended"
+    "plugin:n/recommended", // Uses the recommended rules from eslint-plugin-n (node)
+    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
+    "prettier" // enables eslint-plugin-prettier and displays prettier errors as ESLint errors. (must be the last config in extends array)
   ],
-  "ignorePatterns": ["node_modules/"],
+  "ignorePatterns": ["node_modules/", "artifacts", "dist"],
+  "parser": "@typescript-eslint/parser",
   "parserOptions": {
-    "ecmaVersion": "latest"
+    "project": "./tsconfig.json",
+    "ecmaVersion": 2018,
+    "sourceType": "module" // Allows for the use of imports
+  },
+  "settings": {
+    "import/parsers": {
+      "@typescript-eslint/parser": [".ts"]
+    },
+    "import/resolver": {
+      "typescript": {
+        "alwaysTryTypes": true, // always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
+        "project": "./tsconfig.json"
+      }
+    }
   },
   "rules": {
     "comma-dangle": ["warn", "always-multiline"],
     "curly": ["error", "all"],
     "max-len": ["error", { "code": 120 }],
     "no-nested-ternary": "error",
-    "no-unused-vars": ["warn", { "varsIgnorePattern": "^_unused" }],
-    "padded-blocks": "off"
-  }
+    "padded-blocks": "off",
```

Comment:
Enabling this may be nice for consistency. Right now it works like this:

```
// Both are acceptable
function foo() {
  return true;
}

function bar () {
  return false;
}
```
Personally I prefer the first example

Reply:
[padded-blocks rule](https://eslint.org/docs/latest/rules/padded-blocks) is for enforcing new lines inside curly braces blocks:

```javascript
// padded-blocks: "always"
if (a) {

    b();

}

// padded-blocks: "off"
if (a) {
    b();
}
```

Reply:
Oops, I meant the line below. Misclick I guess. But it is so minor, please don't let this block any merge 👍